apps/dh/api-dh/source/DataHub.WebApi/Modules/MarketParticipant/Actor/ActorOperations.cs

```diff
@@ -13,16 +13,105 @@
+        var associatedActor = user?.GetAssociatedActor()
+            ?? throw new InvalidOperationException("No associated actor found.");
+
+        return client.ActorGetAsync(associatedActor);
+    }
+
+    [Query]
+    public static async Task<ActorDto> GetActorByIdAsync(
+            Guid id,
+            [Service] IMarketParticipantClient_V1 client) =>
+            await client.ActorGetAsync(id);
+
+    [Query]
+    public static async Task<IEnumerable<ActorDto>> GetActorsAsync(
+            [Service] IMarketParticipantClient_V1 client) =>
+            await client.ActorGetAsync();
+
+    [Query]
+    public static async Task<IEnumerable<ActorDto>> GetActorsByOrganizationIdAsync(
+            Guid organizationId,
+            [Service] IMarketParticipantClient_V1 client) =>
+            await client.OrganizationActorAsync(organizationId);
+
+    [Query]
+    public static async Task<IEnumerable<ActorDto>> GetActorsForEicFunctionAsync(
+            EicFunction[]? eicFunctions,
+            [Service] IMarketParticipantClient_V1 client) =>
+            (await client.ActorGetAsync())
+            .Where(x => eicFunctions != null && eicFunctions.Contains(x.MarketRole.EicFunction));
+
+    [Query]
+    public static async Task<AssociatedActors> GetAssociatedActorsAsync(
+        string email,
+        [Service] IMarketParticipantClient_V1 client)
+    {
+        var user = (await client.UserOverviewUsersSearchAsync(
+                1,
+                int.MaxValue,
+                UserOverviewSortProperty.Email,
+                SortDirection.Asc,
+                new UserOverviewFilterDto
+                {
+                    UserStatus = [],
+                    UserRoleIds = [],
+                })).Users
+            .FirstOrDefault(x => string.Equals(email, x.Email, StringComparison.OrdinalIgnoreCase));
+
+        if (user is null)
+        {
+            return new AssociatedActors(email, Enumerable.Empty<Guid>());
+        }
+
+        var associatedActors = await client.UserActorsGetAsync(user.Id);
+
+        return new AssociatedActors(email, associatedActors.ActorIds);
+    }
+
+    [Query]
+    public static async Task<IEnumerable<ActorDto>> FilteredActorsAsync(
+        [Service] IHttpContextAccessor httpContext,
+        [Service] IMarketParticipantClient_V1 client)
+    {
+        if (httpContext.HttpContext == null)
+        {
+            return Enumerable.Empty<ActorDto>();
+        }
+
+        var actors = await client.ActorGetAsync();
+
+        if (httpContext.HttpContext.User.IsFas())
+        {
+            return actors;
+        }
+
+        var actorId = httpContext.HttpContext.User.GetAssociatedActor();
+        return actors.Where(actor => actor.ActorId == actorId);
+    }
+
+    [Query]
+    public static async Task<IEnumerable<SelectionActorDto>> GetSelectionActorsAsync(
+            [Service] IMarketParticipantClient_V1 client) =>
+                await client.QuerySelectionActorsAsync();
```

Comment:
nitpick: indentation
```suggestion
    public static async Task<IEnumerable<SelectionActorDto>> GetSelectionActorsAsync(
        [Service] IMarketParticipantClient_V1 client) =>
        await client.QuerySelectionActorsAsync();
```